app/controllers/tags_controller.rb

```diff
@@ -585,6 +585,7 @@ def build_topic_list_options
   end
 
   def tag_params
-    Array(params[:tags]).concat(Array(@additional_tags))
+    tags = params[:tags].grep(String || Integer)
```

Comment:
This doesn’t work as expected, as @SamSaffron was saying. `String || Integer` is the same as `String` since `String` won’t ever be falsy. So here we’re just grepping for strings. We could probably use `#select` to filter for strings and integers, but integers don’t have a `#downcase` method anyway. Otherwise we could simply cast values to strings with something like this maybe:
```ruby
Array(params[:tags]).map(&:to_s).concat(Array(@additional_tags))
```

Reply:
Thanks, yes that works. I don't think we need to support integers, the use case here is for: 

```
/tag/test.json?tags[]=test1&tags[]=test2
```

Which is already a stretch.